src/ChannelServer/Network/ChannelPacketHandler.cs

```diff
@@ -908,14 +908,35 @@ public void CZ_SKILL_GROUND(ChannelConnection conn, Packet packet)
 
 			// The following code was (currently commented out) is what has been observed from GROUND SKILL packet responses.
 
-			/*
+
 			var packetPosition1 = new Position(x1, y1, z1);
 			var packetPosition2 = new Position(x2, y2, z2);
-			var skillPosition = new Position(x1, y1, z1 - 20);
+			var skillPosition = new Position(40 * cos + x1, y1, 40 * sin + z1);
+
 			var packetDirection = new Direction(cos, sin);
 
 			var skillDirection = new Direction(0.707f, 0.707f);
 
+
+			Log.Debug("SKILL pos1 {0},{1},{2}", x1,y1,z1);
+			Log.Debug("SKILL pos2 {0},{1},{2}", x2, y2,z2);
+
+
+			// Add skill to world
+			// Get map
+			var map = ChannelServer.Instance.World.GetMap(character.MapId);
+
+			SkillActor PESkill = new SkillActor();
+			PESkill.ownerHandle = character.Handle;
+			PESkill.SkillId = skillId;
+			PESkill.Position = skillPosition;
+			PESkill.Init();
+
+			map.AddSkill(PESkill);
```

Comment:
We've talked a little about his yesterday on the chat, in my opinion calling a skill an actor is wrong. A skill can be many things and isn't necessarily an entity on a map. What is an entity would be the skill *effect*.